gtsam/inference/BayesTree-inst.h

```diff
@@ -335,112 +337,85 @@ namespace gtsam {
   }
 
   /* ************************************************************************* */
-  template<class CLIQUE>
-  typename BayesTree<CLIQUE>::sharedBayesNet
-    BayesTree<CLIQUE>::jointBayesNet(Key j1, Key j2, const Eliminate& function) const
-  {
+  // Find the lowest common ancestor of two cliques
+  template <class CLIQUE>
+  static std::shared_ptr<CLIQUE> findLowestCommonAncestor(
```

Comment:
You might want to consider implementing this as a Range Minimum Query for constant time access.

https://en.wikipedia.org/wiki/Lowest_common_ancestor

Reply:
I just moved this. Have no desire to rewrite now.

Reply:
Gotcha, maybe add a comment/TODO for someone else to tackle as an OSS contribution?